app-backend/db/src/test/scala/com/azavea/rf/database/CampaignDaoSpec.scala

```diff
@@ -1152,4 +1159,133 @@ class CampaignDaoSpec
+              insertedCampaign <- CampaignDao
+                .insertCampaign(
+                  campaignCreate.copy(parentCampaignId = None),
+                  user
+                )
+              insertedProjOne <- AnnotationProjectDao.insert(
+                annotationProjectCreateOne.copy(
+                  status = AnnotationProjectStatus.Waiting,
+                  campaignId = Some(insertedCampaign.id)
+                ),
+                user
+              )
+              insertedProjTwo <- AnnotationProjectDao.insert(
+                annotationProjectCreateTwo
+                  .copy(
+                    status = AnnotationProjectStatus.Waiting,
+                    campaignId = Some(insertedCampaign.id)
+                  ),
+                user
+              )
+              _ <- TaskDao.insertTasks(
+                fixupTaskFeaturesCollection(
+                  taskFeaturesCreate,
+                  insertedProjOne,
+                  Some(TaskStatus.Labeled)
+                ),
+                user
+              )
+              _ <- TaskDao.insertTasks(
+                fixupTaskFeaturesCollection(
+                  taskFeaturesCreate,
+                  insertedProjOne,
+                  Some(TaskStatus.Unlabeled)
+                ),
+                user
+              )
+              projTwoTasks <- TaskDao.insertTasks(
+                fixupTaskFeaturesCollection(
+                  taskFeaturesCreate,
+                  insertedProjTwo,
+                  Some(TaskStatus.Unlabeled)
+                ),
+                user
+              )
+              campaignAfterPrjInsert <- CampaignDao.unsafeGetCampaignById(
+                insertedCampaign.id
+              )
+              projOne <- AnnotationProjectDao.unsafeGetById(insertedProjOne.id)
+              projTwo <- AnnotationProjectDao.unsafeGetById(insertedProjTwo.id)
+              _ <- {
+                val task = projTwoTasks.features.head
+                val taskCreateProperties = task.properties.toCreate
+                val taskCreate = TaskFeatureCreate(taskCreateProperties, task.geometry).withStatus(TaskStatus.LabelingInProgress)
+                TaskDao.updateTask(task.id, taskCreate, user)
+              }
+              campaignAfterTaskUpdate <- CampaignDao.unsafeGetCampaignById(
+                insertedCampaign.id
+              )
+            } yield (projOne, projTwo, campaignAfterPrjInsert, campaignAfterTaskUpdate)
+
+            val (projectOne, projectTwo, campaignAfterTaskInsert, campaignAfterTaskUpdate) =
+              updateCampaignProjectIO.transact(xa).unsafeRunSync()
+
+            val campaignLabeled = campaignAfterTaskInsert.taskStatusSummary.getOrElse(TaskStatus.Labeled.toString, 0)
+            val campaignUnlabeled = campaignAfterTaskInsert.taskStatusSummary.getOrElse(TaskStatus.Unlabeled.toString, 0)
+
+            val projectLabeled = projectOne.taskStatusSummary.get.getOrElse(TaskStatus.Labeled.toString, 100)
```

Comment:
What's the `100` magic number? I think the fallback to `100` here will lead to surprising shrinks (shrinking to 100 if the first failure was > 100, shrinking to 0 if it's less maybe) if this test goes awry, so sticking with 0 might serve future us better.

Reply:
Ah, I guess I should have use a random integer for all of the default values in this block. My goal here was to try and prevent a situation where these tests pass (campaign counts = 0 and project counts = 0) if for some reason the task status map is missing the particular keys we're interested in. A world where the key for a particular task status is missing I wanted the tests to fail because that's kind of unexpected - so I chose arbitrary numbers for the two sets of defaults so that they wouldn't equal one another.

Reply:
I think that's fine -- if we end up with this failing in the future and weird shrinks I think it should be clear enough